enzyme/Enzyme/Enzyme.cpp

```diff
@@ -326,6 +327,12 @@ class Enzyme : public ModulePass {
 
       constants.push_back(ty);
 
+      for (auto i = constants.begin(); i != constants.end() && fwdMode; ++i) {
```

Comment:
This is reasonable a reasonable start.

Something to consider (need not be now). you may alternatively want to change the rule for auto detecting diffe type to have this be dup by default for forward mode. This way you can catch if someone passes the out_diff metadata to a forward mode call as an error.

Reply:
Since we've now defined the calling convention for the forward mode, I went ahead and reworked this in 476ef9da884901061662f40ffad92ff28b07d765